ItemTypeOrderedSet.cs

```diff
@@ -76,33 +78,36 @@ public bool RemoveAt(int index)
 
 		public void Save(TagCompound c)
 		{
-			c.Add(_name + Suffix, _items.Select(x => x.type).ToList());
+			c.Add(_name + Suffix3, _items.Select(x => new ItemDefinition(x.type)).ToList());
 		}
 
 		public void Load(TagCompound tag)
 		{
-			IList<TagCompound> list = tag.GetList<TagCompound>(_name);
-			if (list is not null && list.Count > 0)
+			if (tag.GetList<TagCompound>(_name) is { Count: > 0 } listV1) 
 			{
-				_items = list.Select(ItemIO.Load).ToList();
+				_items = listV1.Select(ItemIO.Load).ToList();
 			}
-			else
+			else if (tag.GetList<int>(_name + Suffix) is { Count: > 0 } listV2) 
 			{
-				IList<int> listV2 = tag.GetList<int>(_name + Suffix);
-				if (listV2 is not null)
-					_items = listV2.Select(x =>
-						{
-							if (x >= ItemLoader.ItemCount && ItemLoader.GetItem(x) == null)
-								return null;
-							Item item = new();
-							item.SetDefaults(x);
-							item.type = x;
-							return item;
-						})
-						.Where(x => x is not null)
-						.ToList();
-				else
-					_items = new List<Item>();
+				_items = listV2.Select(x =>
```

Comment:
this code looks pretty old, so we might as well change it while we're at it  
suggested replacement for the LINQ logic:
```cs
_items = listV2.Where(x => x < ItemLoader.ItemCount)  // Unable to reliably restore invalid IDs; just ignore them
    .Select(x => new Item(x))
    .Where(x => !x.IsAir)  // Filters out deprecated items
    .ToList();
```